src/Common/RWLock.cpp

```diff
@@ -169,11 +178,12 @@ RWLockImpl::getLock(RWLockImpl::Type type, const String & query_id, const std::c
     if (rdlock_owner == readers_queue.end() && wrlock_owner == writers_queue.end())
     {
         (type == Read ? rdlock_owner : wrlock_owner) = it_group;  /// SM2: nothrow
```

Comment:
This ternary conditional seems wrong, as this can only be reached if `type == Read`. Otherwise `writers_queue` won't be empty

Reply:
`readers_queue` and `writers_queue` can't be both empty here, at least they must contain `it_group`.
But we don't check those queues for emptyness here, we assign `rdlock_owner` or `wrlock_owner` to `it_group`.
This condition is not wrong.